packages/api/amplify_api/example/integration_test/graphql/iam_test.dart

```diff
@@ -369,6 +369,35 @@ void main({bool useExistingTestUser = false}) {
 
           expect(postFromEvent?.title, equals(title));
         });
+
+        testWidgets(
+            'stream should emit response with error when subscription fails',
+            (WidgetTester tester) async {
+          const document = '''subscription MyInvalidSubscription {
+            onCreateInvalidBlog {
+              id
+              name
+              createdAt
+            }
+          }''';
+          final subscriptionRequest =
+              GraphQLRequest<String>(document: document);
+          final stream = Amplify.API.subscribe(subscriptionRequest,
+              onEstablished: () => fail(
+                    'onEstablished should not be called during failed subscription',
+                  ));
+
+          await expectLater(
+            stream,
+            emits(predicate<GraphQLResponse<String>>(
+              (GraphQLResponse<String> response) =>
+                  response.hasErrors && response.data == null,
+              'Has GraphQL Errors',
+            )),
+          );
+          // Cleanup.
+          await stream.listen((_) {}).cancel();
```

Comment:
This is typically how you'll see it

```suggestion
          await stream.listen(null).cancel();
```

Does this test the issue though? I thought the issue was calling cancel and then receiving an error? This will wait for the error and then call cancel. Should it be like this instead?

```dart
          expect(
            stream,
            emits(predicate<GraphQLResponse<String>>(
              (GraphQLResponse<String> response) =>
                  response.hasErrors && response.data == null,
              'Has GraphQL Errors',
            )),
          );
          // Cleanup.
          await stream.listen(null).cancel();
```

Reply:
Thanks, that helps. I found to repro the issue I had to also add a few second delay after this to give the channel time to throw an error. Otherwise, the error gets thrown in the internal stream and throws error after the test is completed.